pkg/controller/admissionchecks/provisioning/provisioning.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	autoscaling "k8s.io/autoscaler/cluster-autoscaler/apis/provisioningrequest/autoscaling.x-k8s.io/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/util/admissioncheck"
+)
+
+func isProvisioned(pr *autoscaling.ProvisioningRequest) bool {
+	return apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Provisioned)
+}
+
+func isNotProvisioned(pr *autoscaling.ProvisioningRequest) bool {
+	return apimeta.IsStatusConditionFalse(pr.Status.Conditions, autoscaling.Provisioned)
```

Comment:
Function `isProvisioned` will return `false` in two cases:
- if the condition is not present
- if the condition is present and set to `false`
The intention behind `isNotProvisioned` was to return `true` if the condition is present and set to `false`. In case the condition is not present, as you said it will return `false` but that intended behavior.

Reply:
> What will be returned if Provisioned condition is not present? Seems the function will return false, but should true.
I mean, if the Provisioned conditioned is not present, it should be treated in the same way as Provisioned condition is set to false

Reply:
IIUC the difference between Provisioned condition set to `false` and not present, is that in the first case it provides a message with ETA (or will be doing so soon) and if not present the message is not there as well. Am I missing anything?

Reply:
Oh, I see, that makes sense. 
Let's rename the function to `isProvisionedFalse` to avoid confusions.

Reply:
Good point, done!